packages/core/src/Services/socket-general.js

```diff
@@ -307,21 +307,25 @@ export default BinarySocketGeneral;
 const ResponseHandlers = (() => {
     const websiteStatus = response => {
         if (response.website_status) {
-            const is_available = !BinarySocket.isSiteDown(response.website_status.site_status);
-            if (is_available && BinarySocket.getAvailability().is_down) {
-                window.location.reload();
-                return;
-            }
-            const is_updating = BinarySocket.isSiteUpdating(response.website_status.site_status);
-            if (is_updating && !BinarySocket.getAvailability().is_updating) {
-                // the existing connection is alive for one minute while status is updating
-                // switch to the new connection somewhere between 1-30 seconds from now
-                // to avoid everyone switching to the new connection at the same time
-                const rand_timeout = Math.floor(Math.random() * 30) + 1;
+            const { site_status } = response.website_status;
+            const isServerDown = site_status === 'down' || site_status === 'updating';
+
+            // If site is down or updating, connect to WebSocket with an exponentially increasing delay on every attempt
+            if (isServerDown) {
+                let reconnectionCounter = +(window.localStorage.getItem(RECONNECTION_COUNTER_KEY) || 1);
+                const reconnectionDelay = Math.min(Math.pow(2, reconnectionCounter + 9), 600000);
+
                 window.setTimeout(() => {
+                    window.localStorage.setItem(RECONNECTION_COUNTER_KEY, (++reconnectionCounter).toString());
```

Comment:
could we use global variable instead of localStorage?
to mitigate the case when e.g. someone might reload the page during backoff (basically to ensure that whatever happens, reloading the webpage will clean the state, so it will never end up in persistently bad state even if something would go completely wrong)

Reply:
i was trying to persist the counter if the user refreshes, but i might miss some edge cases u mention, gonna see if this is can be improved

Reply:
Maybe can set cookies for this counter to persist the counter. Because during inaccessibility a novice user will try to refresh or hard reload the page multiple times so the counter will be reset to default value 🤔

Reply:
I think pretty much standard in backoff logic is to not persist the state. If state gets broken, user might be left with broken state. The edge case I'm thinking about is mild (extra page refresh), but basically playing with reloading based on state which is persisting is like playing with fire.

Reply:
@wojciech-deriv  i think i'll take the advice and be safe with this - using global variable instead

Reply:
This is resolved. Have a look @wojciech-deriv